### PR TITLE
Initial commit for Focus Sequence

### PR DESCRIPTION
**Pull request by isotuela**
Hi Patrick, here's the initial commit for the notebook with the parabolic focus sequence data acquisition. Thank you for your comments!

### CODE REVIEW

File: procedures/auxtel/observation_procedures/Parabolic_Focus_Sequence.ipynb
Comment (patrickingraham) [inline procedures/auxtel/observation_procedures/Parabolic_Focus_Sequence.ipynb]: <p>Instantiate</p><p> </p><p>I think it would be good to write a sentence or two about the reasoning for instantiating the CWFS script. It seems a bit odd to use a CWFS script to do a parabolic focusing :)</p><p>Consider something like:</p><p>In many cases it is desirable to perform a CWFS focus sequence for comparison. Therefore, we instantiate this script and provide the option to the user after the Parabolic focus data acquisition.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/lsst-ts/ts_notebooks/pull/88/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='RESOLVED' data-cellIndex='8'/>

Comment (patrickingraham) [inline procedures/auxtel/observation_procedures/Parabolic_Focus_Sequence.ipynb]: <pre class="ql-syntax" spellcheck="false">Line #1.    Filter_to_focus = 'SDSSg'
</pre><p>Maybe make this SDSSr since g probably won't go back in the instrument</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/lsst-ts/ts_notebooks/pull/88/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='RESOLVED' data-cellIndex='20'/>

Comment (patrickingraham) [inline procedures/auxtel/observation_procedures/Parabolic_Focus_Sequence.ipynb]: <p>you could use an assertion here if you want a more official check</p><p> </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/lsst-ts/ts_notebooks/pull/88/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='RESOLVED' data-cellIndex='23'/>

Comment (patrickingraham) [inline procedures/auxtel/observation_procedures/Parabolic_Focus_Sequence.ipynb]: <p>consider merging these two cells. Something like</p><pre class="ql-syntax" spellcheck="false">test = await script.atcs.rem.ataos.evt_correctionEnabled.aget())

if not (test.m1 and test.atspectrograph and test.m1):
  await script.atcs.rem.ataos.cmd_enableCorrection.set_start(m1=True, hexapod=True, atspectrograph=True)
</pre><p> </p><p>you also might be able to put it in one line using a tertiary statement.... but i'm not sure.</p><p> </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/lsst-ts/ts_notebooks/pull/88/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='RESOLVED' data-cellIndex='24'/>

Comment (patrickingraham) [inline procedures/auxtel/observation_procedures/Parabolic_Focus_Sequence.ipynb]: <p>Consider:</p><p>Declare a target name</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/lsst-ts/ts_notebooks/pull/88/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='RESOLVED' data-cellIndex='27'/>

Comment (patrickingraham) [inline procedures/auxtel/observation_procedures/Parabolic_Focus_Sequence.ipynb]: <p>Consider:</p><p>Query for a target at a user-defined azimuth and elevation</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/lsst-ts/ts_notebooks/pull/88/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='RESOLVED' data-cellIndex='29'/>

Comment (patrickingraham) [inline procedures/auxtel/observation_procedures/Parabolic_Focus_Sequence.ipynb]: <p>Consider adding a line after this with the code to offset the telescope a bit. If the star falls in a bad place then we'll want to move it a little.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/lsst-ts/ts_notebooks/pull/88/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='RESOLVED' data-cellIndex='34'/>

Comment (patrickingraham) [inline procedures/auxtel/observation_procedures/Parabolic_Focus_Sequence.ipynb]: <p>I don't think the "-5x to 5x" is quite clear enough.</p><p>Consider:</p><p>Acquire data in steps of <code>z_offset_step</code> in the range of <code>z_offset_start</code> to <code>z_offset_end</code> relative to the current focus position.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/lsst-ts/ts_notebooks/pull/88/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='RESOLVED' data-cellIndex='35'/>

Comment (patrickingraham) [inline procedures/auxtel/observation_procedures/Parabolic_Focus_Sequence.ipynb]: <pre class="ql-syntax" spellcheck="false">Line #12.        print(f"Total z offset = {step:.3f}")
</pre><p>You're using the logger above no? Maybe make these log statements? I'm unsure.... your call.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/lsst-ts/ts_notebooks/pull/88/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='RESOLVED' data-cellIndex='37'/>

Comment (patrickingraham) [inline procedures/auxtel/observation_procedures/Parabolic_Focus_Sequence.ipynb]: <pre class="ql-syntax" spellcheck="false">Line #14.        exposure = await latiss.take_engtest(
</pre><p>it would be great to populate the REASON header here. You could call it "Parabolic_focus_sequence" or something. I'm not sure if there is a list of defined labels already or not.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/lsst-ts/ts_notebooks/pull/88/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='RESOLVED' data-cellIndex='37'/>

Comment (patrickingraham) [inline procedures/auxtel/observation_procedures/Parabolic_Focus_Sequence.ipynb]: <pre class="ql-syntax" spellcheck="false">Line #22.    # Put offset back to original position
</pre><p>I think I wrote this... but it could be better</p><p>Consider:</p><p>"Move back to the original focus offset position"</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/lsst-ts/ts_notebooks/pull/88/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='RESOLVED' data-cellIndex='37'/>

Comment (patrickingraham) [inline procedures/auxtel/observation_procedures/Parabolic_Focus_Sequence.ipynb]: <p>You might say why you're doing this... which is surely to run the analysis notebook?</p><p>We should really write this to the LFA... but I'm not sure there is an elegant method to do that yet... </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/lsst-ts/ts_notebooks/pull/88/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='RESOLVED' data-cellIndex='39'/>

Comment (patrickingraham) [inline procedures/auxtel/observation_procedures/Parabolic_Focus_Sequence.ipynb]: <p>I think you want to make it clear that this is optional!</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/lsst-ts/ts_notebooks/pull/88/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='RESOLVED' data-cellIndex='42'/>

Comment (patrickingraham) [inline procedures/auxtel/observation_procedures/Parabolic_Focus_Sequence.ipynb]: <pre class="ql-syntax" spellcheck="false">Line #1.    configuration = yaml.safe_dump({"filter": Filter_to_focus, 
</pre><p>Variables should all be lowercase <code>filter_to_focus</code>  (python pep8 standard) </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/lsst-ts/ts_notebooks/pull/88/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='RESOLVED' data-cellIndex='46'/>

**Comment by patrickingraham** [comment]
Thanks Ioana.  A nice first cut! 
Can you remind me if you are making a second analysis notebook as well?

File: procedures/auxtel/observation_procedures/Parabolic_Focus_Sequence.ipynb
Comment (isotuela) [reply procedures/auxtel/observation_procedures/Parabolic_Focus_Sequence.ipynb]: <p>Thanks! I also changed the variable name to lowercase to reflect python pep8 standard. </p>

Comment (isotuela) [reply procedures/auxtel/observation_procedures/Parabolic_Focus_Sequence.ipynb]: <p>Cheers. Added. </p><p> </p>

Comment (isotuela) [reply procedures/auxtel/observation_procedures/Parabolic_Focus_Sequence.ipynb]: <p>Definitely better. Two cells merged. Replaced one of the <code>test.m1</code> by <code>test.hexapod</code></p>

Comment (isotuela) [reply procedures/auxtel/observation_procedures/Parabolic_Focus_Sequence.ipynb]: <p>Done</p>

Comment (isotuela) [reply procedures/auxtel/observation_procedures/Parabolic_Focus_Sequence.ipynb]: <p>Done</p>

Comment (isotuela) [reply procedures/auxtel/observation_procedures/Parabolic_Focus_Sequence.ipynb]: <p>Sure,  useful! </p>

Comment (isotuela) [reply procedures/auxtel/observation_procedures/Parabolic_Focus_Sequence.ipynb]: <p>Considered and modified. </p>

Comment (isotuela) [reply procedures/auxtel/observation_procedures/Parabolic_Focus_Sequence.ipynb]: <p>Changed <code>print</code> statements to <code>logger.info</code>     </p>

Comment (isotuela) [reply procedures/auxtel/observation_procedures/Parabolic_Focus_Sequence.ipynb]: <p>Agree and modified. </p>

Comment (isotuela) [reply procedures/auxtel/observation_procedures/Parabolic_Focus_Sequence.ipynb]: <p>Sounds good. </p>

Comment (isotuela) [reply procedures/auxtel/observation_procedures/Parabolic_Focus_Sequence.ipynb]: <p>Done</p>

Comment (isotuela) [reply procedures/auxtel/observation_procedures/Parabolic_Focus_Sequence.ipynb]: <p>Done. </p><p> </p>

Comment (isotuela) [reply procedures/auxtel/observation_procedures/Parabolic_Focus_Sequence.ipynb]: <p>Instead of saving a file, I've sent the sequence information into the script log to show up in the EFD <code>lsst.sal.Script.logevent_logMessage</code>  <span class="ql-cursor">﻿</span></p>

Comment (isotuela) [reply procedures/auxtel/observation_procedures/Parabolic_Focus_Sequence.ipynb:1]: <p><code>script_log.log.info(f'END -- Focus Determination {filter_to_focus}+{grating_to_focus} \n'</p><p>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;f'Sequences -- {parabolic_focus_data} \n'</p><p>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;f'Time UTC -- {astropy.time.Time(astropy.time.Time.now())}') </code> </p>

Comment (patrickingraham) [reply procedures/auxtel/observation_procedures/Parabolic_Focus_Sequence.ipynb]: <p>I think that's a good idea for the moment. </p>

**Review by patrickingraham — APPROVED** [review]
Much improved! Thanks for making the changes.
Please be sure to clear all cells before merging. It looks like a few were run.

**Comment by isotuela** [comment]
Thank you!

**Pull request merged**
